framework/Metrics/CDFAreaDifference.py

```diff
@@ -31,23 +31,38 @@ class CDFAreaDifference(Metric):
   """
     Metric to compare two datasets using the CDF Area Difference.
   """
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, NOne
+    """
+    Metric.__init__(self)
+    # If True the metric needs to be able to handle (value,probability) where value and probability are lists
+    self.acceptsProbability = True
+    # If True the metric needs to be able to handle a passed in Distribution
+    self.acceptsDistribution = True
+
   def _localReadMoreXML(self,xmlNode):
     """
       Method that reads the portion of the xml input that belongs to this specialized class
       and initializes internal parameters
       @ In, xmlNode, xml.etree.Element, Xml element node
       @ Out, None
     """
-    self.acceptsProbability = True
-    self.acceptsDistribution = True
+    pass
 
-  def distance(self,x,y,**kwargs):
+  def __evaluateLocal__(self, x, y, weights = None, axis = 0, **kwargs):
     """
-      Calculates the CDF Area Difference between two datasets.
-      @ In, x, something that can be converted into a CDF
-      @ In, y, something that can be converted into a CDF
-      @ In, kwargs, ignored.
-      @ Out, value, float, CDF Area Difference
+      This method computes difference between two points x and y based on given metric
+      @ In, x, numpy.array or instance of Distributions.Distribution, array containing data of x,
+        or given distribution.
+      @ In, y, numpy.array, or instance of Distributions.Distribution, array containing data of y,
+        or given distribution.
+      @ In, weights, None or array_like (numpy.array or list), not used in this metric
+      @ In, axis, integer, default is 0, not used for this metric.
+      @ In, kwargs, dictionary of parameters characteristic of each metric
```

Comment:
@ In, kwargs, dict, dictionary of parameters characteristic of each metric

Reply:
fixed.